### ISSUE TITLE
RMG 3.1.0 cannot reproduce the result of RMG 3.0.0

### ISSUE THREAD

**Issue by utopianf** [ReactionMechanismGenerator/RMG-Py#2313]
### Bug Description
We are trying to reproduce the paper on [Catalytic Partial Oxidation of Methane](https://chemrxiv.org/engage/chemrxiv/article-details/60c75875469df42761f45666) by RMG 3.1.0 and RMG 3.0.0. The result of RMG 3.0.0 seems the same as the paper which CH4 mostly converts into CO2 (CH4 -> CH3X -> CH2X -> CHX -> CX -> COX -> CO2X -> CO2). However, RMG 3.1.0 shows CH4 converts into C2H6 (CH4 -> CH4X -> CH3X -> C2H6), which cannot reproduce the result of the paper.

By looking into the output.html, I found the one of the kinetics of the reactions (Surface_Adsorption_Dissociative) is different between RMG 3.0.0 and RMG 3.1.0. But [the rule of this reaction](https://github.com/ReactionMechanismGenerator/RMG-database/blob/main/input/kinetics/families/Surface_Adsorption_Dissociative/rules.py#L9-L23) is the same between RMG-database 3.0.0 and RMG-database 3.1.0. 

Is this a bug? Or could someone please provide a reasonable explanation of the huge different kinetics between versions?

output.html of RMG 3.1.0:
![image](https://user-images.githubusercontent.com/8581846/173197265-ba01da1a-ae5e-4598-8fbd-acf7c394b865.png)

output.html of RMG 3.0.0:
![image](https://user-images.githubusercontent.com/8581846/173197295-e3caa3db-05fb-44f2-822a-a42d3f65a3ca.png)

### How To Reproduce
1. Download [the zip file](https://pubs.acs.org/doi/suppl/10.1021/acscatal.0c04100/suppl_file/cs0c04100_si_002.zip) which includes input.py.
2. `rmg.py input.py` by both RMG 3.0.0 and RMG 3.1.0.

### Expected Behavior
CH4 should finally convert into CO2.

### Installation Information
Describe your installation method and system information.
 - OS (include version if known): WSL2 on Windows 10
 - Installation method: installation from binary, with anaconda
 - RMG version information:
   
   - RMG-Py: 3.1.0 and 3.0.0
   - RMG-database: 3.1.0 and 3.0.0

### Additional Context
We also make some illustration by graphviz to see what happened during the reaction.

RMG 3.1.0:
![image](https://user-images.githubusercontent.com/8581846/173197570-1470e507-a70f-4875-b984-dc9ca25af3b5.png)

RMG 3.0.0:
![image](https://user-images.githubusercontent.com/8581846/173197580-ea51d795-023e-4f31-a5e3-24d0c229ca1c.png)

**Comment by utopianf** [comment]
@bjkreitz Thanks for looking into this and sorry for the late reply! The problem is still open, and we use metal Rh(111). The definition is as below
```
catalystProperties( # default values for Rh(111)
    bindingEnergies = {
                       'C':(-6.568, 'eV/molecule'),
                       'O':(-4.610, 'eV/molecule'),
                       'N':(-4.352, 'eV/molecule'),
                       'H':(-2.479, 'eV/molecule'),
                       },
    surfaceSiteDensity=(2.72e-9, 'mol/cm^2'),
)
```

**Comment by ChrisBNEU** [comment]
Per our discussion at the 2023 rmg training, there are a number of new families , and a number of new rate rules that we've added. I think the best way to deal with this is by limiting the following within your mechanism: 
- the amount of carbon atoms (<=C3 is recommended, since we do not have a lot of thermo for larger. for cpox specifying C1 might be the right way to go, I'm not sure if there are any c2 intermediates)
- the amount of surface sites (<=2 is recommended, we do not have kinetics beyond bidentates)

For the cpox model, we have added a lot of new families and rules that will affect this model. To reproduce the exact results of the paper you will need to use the iteration of the database and rmg-py that were specified (I believe both of those are on zenodo)

To address the root cause of the problem, we need to implement regression tests for our surface mechanisms. Currently we do not have a regression test for RMG-cat mechanisms, but this is addressed in #2004 and [#85](https://github.com/ReactionMechanismGenerator/RMG-tests/pull/85).
